bftengine/src/bcstatetransfer/SourceSelector.hpp

```diff
@@ -73,22 +85,34 @@ class SourceSelector {
 
   void addPreferredReplica(uint16_t replicaId) { preferredReplicas_.insert(replicaId); }
 
+  uint16_t currentPrimary() { return currentPrimary_; }
+
+  void removePreferredReplica(uint16_t replicaId) { preferredReplicas_.erase(replicaId); }
+
   uint16_t numberOfPreferredReplicas() const { return static_cast<uint16_t>(preferredReplicas_.size()); }
 
-  bool isPreferred(uint16_t replicaId) const { return preferredReplicas_.count(replicaId) != 0; }
+  bool isPreferredSourceId(uint16_t replicaId) const { return preferredReplicas_.count(replicaId) != 0; }
+
+  bool isValidSourceId(uint16_t replicaId) const { return allOtherReplicas_.count(replicaId) != 0; }
 
   uint16_t currentReplica() const { return currentReplica_; }
 
   void onReceivedValidBlockFromSource();
 
   const std::vector<uint16_t> &getActualSources() { return actualSources_; }
 
+  void updateCurrentPrimary(uint16_t newPrimaryReplicaId);
+
+  uint16_t minPrePrepareMsgsForPrimaryAwarness() { return minPrePrepareMsgsForPrimaryAwarness_; }
```

Comment:
The impcat of the UT should be minimal on the production.
Here you create unneded getters, and potentially many more in the future - it looks akward if the reader is unaware of the reason. I think it should be removed,  the best idea is to add BcStTest as a friend of this class.